src/serialize/window.js

```diff
@@ -45,6 +45,10 @@ function getSerializedWindow(winPromise : ZalgoPromise<CrossDomainWindowType>, {
     return {
         id,
         getType: () => winPromise.then(win => {
+            if (isWindowClosed(win)) {
+                return WINDOW_TYPE.CLOSED;
+            }
+
             return getOpener(win) ? WINDOW_TYPE.POPUP : WINDOW_TYPE.IFRAME;
```

Comment:
I'm wondering... can we do this check when the window is first serialized, rather than on-demand? That ought to fix this without introducing a new window type.